src/main/java/org/jenkinsci/plugins/workflow/support/concurrent/Futures.java

```diff
@@ -45,6 +41,7 @@
  *
  * @author Guava
  */
+@Deprecated
```

Comment:
This class is used from `workflow-cps`, `workflow-cps-checkpoint`, `workflow-job`, and `workflow-support`. Deprecating it so that these plugins can prefer the upstream version once core is on a recent version of Guava and the plugins' baselines have been updated.

Reply:
Which could now be whenever we like, right?

Also some of these methods could be deprecated in favor of Java Platform equivalents, like `CompletableFuture`.

Reply:
Are we okay with doing that in a separate PR?

Reply:
Sure.